resource/abstractions/auditing_rules.md

```diff
@@ -0,0 +1,478 @@
+```
+
+#### 📄 Description
+
+Clear any rules defined within the **Audit Rules** for a general resource profile.
+
+#### 📥 Parameters
+* `resource`<br>
+  The **general resource profile** for which RACF should change the audit rules.
+
+* `class_name`<br>
+  The name of the **class** the general resource profile being changed belongs to.
+
+#### 📤 Returns
+* `Union[dict, bytes]`<br>
+  Returns a **Security Result Steps dictionary** or **Concatenated Security Request XML bytes** if the `ResourceAdmin.generate_requests_only` class attribute is set to `True`.
+
+#### ❌ Raises
+* `SecurityRequestError`<br>
+  Raises `SecurityRequestError` when the **SAF Return Code** of a **Security Result** is equal to `4`.
+* `DownstreamFatalError`<br>
+  Raises `DownstreamFatalError` when the **SAF Return Code** of a **Security Result** is greater than `4`.
+* `AlterOperationError`<br>
+  Raises `AlterOperationError` when the **general resource profile** cannot be altered because it does **NOT** exist.
+
+#### 💻 Example
+
+###### Python REPL
+```python
+>>> from pyracf import ResourceAdmin
+>>> resource_admin = ResourceAdmin()
+>>> resource_admin.clear_all_audit_rules("TESTING","ELIJTEST")
+{"step1":{"securityResult":{"resource":{"name":"TESTING","class":"ELIJTEST","operation":"set","requestId":"ResourceRequest","info":["Definition exists. Add command skipped due  to precheck option"],"commands":[{"safReturnCode":0,"returnCode":0,"reasonCode":0,"image":"RALTER  ELIJTEST             (TESTING)  AUDIT( NONE    )","messages":["ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."]}]},"returnCode":0,"reasonCode":0}}}
+```
+
+###### Security Result Steps Dictionary as JSON
+```json
+{
+  "step1": {
+    "securityResult": {
+      "resource": {
+        "name": "TESTING",
+        "class": "ELIJTEST",
+        "operation": "set",
+        "requestId": "ResourceRequest",
+        "info": [
+          "Definition exists. Add command skipped due  to precheck option"
+        ],
+        "commands": [
+          {
+            "safReturnCode": 0,
+            "returnCode": 0,
+            "reasonCode": 0,
+            "image": "RALTER  ELIJTEST             (TESTING)  AUDIT( NONE    )",
+            "messages": [
+              "ICH11009I RACLISTED PROFILES FOR ELIJTEST WILL NOT REFLECT THE UPDATE(S) UNTIL A SETROPTS REFRESH IS ISSUED."
+            ]
+          }
+        ]
+      },
+      "returnCode": 0,
+      "reasonCode": 0
+    }
+  }
+}
+```
+
+## `ResourceAdmin.overwrite_audit_by_attempt()`
+
+```python
+def overwrite_audit_by_attempt(
+    self,
+    resource: str,
+    class_name: str,
+    successes: Union[str, None] = None,
+    failure: Union[str, None] = None,
+    all: Union[str, None] = None,
+) -> Union[dict, bytes]:
+```
+
+#### 📄 Description
+
+Clear any rules defined within the **Audit Rules** for a general resource profile and replace them with `Success`, `Failure`, or `All` rule(s).
```

Comment:
```markdown
Remove all currently defined **Auditing Rules** for a general resource profile and replace them with the new **Auditing Rules** specified.
```

Reply:
agree that this is cleaner. Applied here and to `overwrite_audit_by_access_level`